pyretailscience/revenue_tree.py

```diff
@@ -87,90 +79,55 @@ def __init__(
+
         if group_col is not None:
-            df[group_col] = pd.Categorical(df[group_col])
-            p1_group = df[p1_index].groupby(group_col, observed=False)
-            p2_group = df[p2_index].groupby(group_col, observed=False)
-            p1_df = p1_group.agg(
-                customers=(cols.customer_id, "nunique"),
-                transactions=(cols.transaction_id, "nunique"),
-                spend=(cols.unit_spend, "sum"),
-            ).rename(
-                columns={
-                    "customers": cols.agg_customer_id,
-                    "transactions": cols.agg_transaction_id,
-                    "spend": cols.agg_unit_spend,
-                },
-            )
-            p2_df = p2_group.agg(
-                customers=(cols.customer_id, "nunique"),
-                transactions=(cols.transaction_id, "nunique"),
-                spend=(cols.unit_spend, "sum"),
-            ).rename(
-                columns={
-                    "customers": cols.agg_customer_id,
-                    "transactions": cols.agg_transaction_id,
-                    "spend": cols.agg_unit_spend,
-                },
-            )
-            if cols.unit_qty in df.columns:
-                p1_df[cols.agg_unit_qty] = p1_group[cols.unit_qty].sum()
-                p2_df[cols.agg_unit_qty] = p2_group[cols.unit_qty].sum()
-        else:
-            p1_df = df[p1_index]
-            p2_df = df[p2_index]
-            p1_df = pd.DataFrame(
-                {
-                    cols.agg_customer_id: p1_df[cols.customer_id].nunique(),
-                    cols.agg_transaction_id: p1_df[cols.transaction_id].nunique(),
-                    cols.agg_unit_spend: p1_df[cols.unit_spend].sum(),
-                },
-                index=["p1"],
-            )
-            p2_df = pd.DataFrame(
-                {
-                    cols.agg_customer_id: p2_df[cols.customer_id].nunique(),
-                    cols.agg_transaction_id: p2_df[cols.transaction_id].nunique(),
-                    cols.agg_unit_spend: p2_df[cols.unit_spend].sum(),
-                },
-                index=["p2"],
-            )
-            if cols.unit_qty in df.columns:
-                p1_df[cols.agg_unit_qty] = df[p1_index][cols.unit_qty].sum()
-                p2_df[cols.agg_unit_qty] = df[p2_index][cols.unit_qty].sum()
+            p1_df = p1_df.sort_values(by=group_col)
+            p2_df = p2_df.sort_values(by=group_col)
 
         new_p1_index = [True] * len(p1_df) + [False] * len(p2_df)
         new_p2_index = [not i for i in new_p1_index]
 
-        return pd.concat([p1_df, p2_df]), new_p1_index, new_p2_index
+        result_df = pd.concat([p1_df, p2_df], ignore_index=True)
+
+        if group_col is not None:
+            result_df.set_index(group_col, inplace=True)
+            result_df.index = pd.CategoricalIndex(result_df.index)
+        else:
+            result_df.index = ["p1", "p2"]
```

Comment:
Tiny comment. The below reduces double negatives making the code a little easier to read. Optional
```suggestion
        if group_col is None:
            result_df.index = ["p1", "p2"]        
        else:
            result_df.set_index(group_col, inplace=True)
            result_df.index = pd.CategoricalIndex(result_df.index)
```